### PR TITLE
User portal on steroids 💉 

### PR DESCRIPTION
**Pull request by frimpongopoku**
####  Summary / Highlights
Depends on [PR](https://github.com/massenergize/api/pull/979)

#### Requirements (place an `x` in each `[ ]`)
* [X] I've read and understood the [Contributing Guidelines](/blob/main/.github/contributing.md) and have done my best effort to follow them.
* [ ] I've tested my changes manually.
* [ ] I've added unit tests to my PR.

##### Transparency ([Project board](https://github.com/orgs/massenergize/projects/7/views/2))
* [X] I've given my PR a meaningful title.
* [ ] I linked this PR to the relevant issue.
* [ ] I moved the linked issue from the "Sprint backlog" to "In progress" when I started this.
* [ ] I moved the linked issue to "QA Verification" now that it is ready to merge.

# More Information 
This PR speeds up the user portal significantly. We have now changed the api request structure from loading all the data upfront to fetching only necessary data per page

### CODE REVIEW

**Review by archx3 — APPROVED** [review]

File: src/AppRouter.js [hunk 1]
```diff
@@ -219,135 +224,178 @@ class AppRouter extends Component {
         apiCall("communities.features.flags.list", body),
         apiCall("home_page_settings.info", body),
         apiCall("menus.list", body),
```
Comment (abdullai-t) [inline src/AppRouter.js:226]: This has to be changed to match the new endpoint

File: src/AppRouter.js [hunk 2]
```diff
@@ -390,71 +451,76 @@ class AppRouter extends Component {
...
+  //       case "/events":
+  //         return this.state.pagesEnabled.eventsPage;
+  //       case "/home":
+  //         return true;
+  //       default:
+  //         return item.children ? item.children.length > 0 : true;
+  //     }
+  //   });
+  //   return this.addPrefix(menu);
+  // }
```
Comment (abdullai-t) [inline src/AppRouter.js:506]: remove if not needed

File: src/AppRouter.js [hunk 3]
```diff
@@ -219,135 +224,178 @@ class AppRouter extends Component {
...
+    //       this.props.reduxLoadCommunitiesStats(communityStatsResponse.data);
+    //       this.props.reduxLoadEquivalences(eqResponse.data);
+    //       this.props.reduxLoadCommunities(listOfCommunitiesResponse.data);
+    //     })
+    //     .catch((err) => {
+    //       this.setState({ error: err });
+    //       console.log(err);
+    //     });
+    // }
   }
```
Comment (abdullai-t) [inline src/AppRouter.js:399]: remove

File: src/AppRouter.js [hunk 4]
```diff
@@ -219,135 +224,178 @@ class AppRouter extends Component {
...
+          //     contactUsPage: contactUsPageResponse.data.is_published,
+          //     donatePage: donatePageResponse.data.is_published,
+          //     eventsPage: eventsPageResponse.data.is_published,
+          //     impactPage: impactPageResponse.data.is_published,
+          //     vendorsPage: vendorsPageResponse.data.is_published,
+          //     testimonialsPage: testimonialsPageResponse.data.is_published,
+          //     teamsPage: teamsPageResponse.data.is_published,
+          //   },
+          //   prefix,
+          // });
```
Comment (abdullai-t) [inline src/AppRouter.js:254]: remove

**Review by abdullai-t — APPROVED** [review]
LGTM ✅

**Review by archx3 — APPROVED** [review]

**Review by abdullai-t — APPROVED** [review]

**Review by archx3 — APPROVED** [review]

**Pull request merged**